File: python-code/broadcast.py
from mpi4py import MPI

comm = MPI.COMM_WORLD
rank = comm.rank

if (rank == 0):
    data = 1000
else:
    data = None

# all processes will get the variable data from process 0
data = comm.bcast(data, root=0)

print("Process", rank, "has data", data)

File: c-code/broadcast.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char** argv) {
    int rank, size;
    int data = 42;

    MPI_Init(&argc, &argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Comm_size(MPI_COMM_WORLD, &size);

    if (rank == 0) {
        // Broadcast data from root process to all other processes
        MPI_Bcast(&data, 1, MPI_INT, 0, MPI_COMM_WORLD);
    } else {
        // Receive data broadcasted from root process
        MPI_Bcast(&data, 1, MPI_INT, 0, MPI_COMM_WORLD);
        printf("Process %d received data: %d\n", rank, data);
    }

    MPI_Finalize();
    return 0;
}
